### PR TITLE
Remove trash, and use toolbox instead

### PR DESCRIPTION
**Pull request by breville**
- Supports LTR and RTL
- Works on toolbox, non-toolbox and function editor.
- Tested on IE9 and desktop Chrome

### CODE REVIEW

**Comment by breville** [comment]
Not sure if it's feature complete, but this work is probably ready for a first review, thanks @Bjvanminnen

File: blockly-core/core/ui/block_space/flyout.js [hunk 1]
```diff
@@ -124,17 +124,28 @@ Blockly.Flyout.prototype.onResizeWrapper_ = null;
  * Creates the flyout's DOM.  Only needs to be called once.
  * @return {!Element} The flyout's SVG group.
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/block_space/flyout.js]: Add an @param tag for new param

File: blockly-core/core/ui/block_space/flyout.js [hunk 2]
```diff
@@ -124,17 +124,28 @@ Blockly.Flyout.prototype.onResizeWrapper_ = null;
...
   </g>
   */
-  this.svgGroup_ = Blockly.createSvgElement('g', {}, null);
+  this.svgGroup_ = Blockly.createSvgElement('g', {'class': 'svgFlyoutGroup'}, null);
   this.svgBackground_ = Blockly.createSvgElement('path',
       {'class': 'blocklyFlyoutBackground'}, this.svgGroup_);
   this.svgGroup_.appendChild(this.blockSpace_.createDom());
+
+  // Add a trashcan.
+  if (! insideToolbox) {
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/block_space/flyout.js]: no space after !

File: blockly-core/core/utils/utils.js [hunk 3]
```diff
@@ -388,10 +388,22 @@ Blockly.convertCoordinates = function(x, y, svg, toSvg) {
  * @return {!Object} Object with .x and .y properties.
  */
 Blockly.mouseToSvg = function(e, opt_svgParent) {
+  return Blockly.mouseCoordinatesToSvg(e.clientX, e.clientY, opt_svgParent || Blockly.topMostSVGParent(e.target));
```
Comment (Bjvanminnen) [inline blockly-core/core/utils/utils.js]: i suspect this is over 80 characters?  Also true of a couple of other places in this PR. We're not super strict about it, but it's good to try to stay under.

**Comment by Bjvanminnen** [comment]
Couple minor nits, but otherwise looks good :)

**Comment by breville** [comment]
@Bjvanminnen A couple new commits to this PR which probably deserve individual review:

https://github.com/code-dot-org/code-dot-org/commit/ea25b63d3c4df86762d4f8a26f3f44a9db37e916
https://github.com/code-dot-org/code-dot-org/commit/477bb4b1706dcc02e1be502c9973e32346b1929c

**Comment by bcjordan** [comment]
Re: `Blocks dragged over toolbox now go over it`, would this cause some weird looking behavior if horizontal scrolling were to be re-enabled? I.e., when scrolling the workspace, leftmost blocks would float above the toolbox?

I don't think we use that or have plans to use that again, so probably OK, but worth remembering if we get asked to re-enable

**Comment by Bjvanminnen** [comment]
Left some comments about naming, but otherwise lgtm

**Pull request merged**